### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by lwr20**
Requires https://github.com/projectcalico/libcalico/pull/163 to be merged before they will pass.

### CODE REVIEW

**Comment by tomdee** [comment]
What do we need to do to make this pass on semaphore

File: tests/st/test_calicoctl.py [hunk 1]
```diff
@@ -0,0 +1,1085 @@
...
+                out = host.calicoctl(
+                    "get %s --output=yaml" % testdata['kind'])
+                output = yaml.safe_load(out)
+                assert output == [], "Testdata has left data in datastore " \
+                                     "instead of being completely " \
+                                     "rejected:\n" \
+                                     "Injected: %s\n" \
+                                     "Got back: %s" % (testdata, output)
+
+            host.writefile("testfile.yaml", testdata)
```
Comment (matthewdupre) [inline tests/st/test_calicoctl.py]: DockerHost has no attribute writefile.  Should this be self.writefile() instead?

**Comment by matthewdupre** [comment]
Tests are failing because the libcalico PR is required.

**Comment by matthewdupre** [comment]
A few test failures to look at.

**Comment by matthewdupre** [comment]
Closing in favour of #1269

**Pull request closed without merging**